Tropos/Controllers/SettingsController.swift

```diff
@@ -0,0 +1,40 @@
+import Foundation
+
+@objc(TRSettingsController) final class SettingsController: NSObject {
+    private let locale: NSLocale
+
+    var unitSystem: UnitSystem {
+        get {
+            let rawUnitSystem = NSUserDefaults.standardUserDefaults().integerForKey(TRSettingsUnitSystemKey)
+            return UnitSystem(rawValue: rawUnitSystem)!
+        }
+        set {
+            NSUserDefaults.standardUserDefaults().setInteger(newValue.rawValue, forKey: TRSettingsUnitSystemKey)
+        }
+    }
+
+    init(locale: NSLocale) {
+        self.locale = locale
+    }
+
+    convenience override init() {
+        self.init(locale: .autoupdatingCurrentLocale())
+    }
+
+    func registerSettings() {
+        registerUnitSystem()
+        registerLastVersion()
```

Comment:
Would you mind elaborating? I seem to remember seeing conversations recently of a change (improvement) in behaviour. Perhaps in a recent version of iOS.